fix! : [DEPR] Remove deprecated CookieNameChange middleware

ktyagiapphelix2u opened the pull request:
Problem Statement:
This PR removes the CookieNameChange middleware that was intended to be temporary with a target removal date of October 1, 2021. This middleware was designed for a transitional cookie migration pattern that is no longer needed.

Solution:
Complete removal of:
- The middleware registration from LMS and CMS settings files
- The middleware implementation file and associated tests

Technical Details:
The middleware was likely inactive in all installations since:
- COOKIE_NAME_CHANGE_ACTIVATE defaulted to False
- The code contained comments indicating it was temporary
- The target removal date (Oct 2021) was over 3 years ago

Testing Performed:
- Verified no active configuration sets COOKIE_NAME_CHANGE_ACTIVATE to True
- Ran tests to ensure removal doesn't break any functionality
- Verified the middleware is not referenced elsewhere in the codebase

robrap commented on cms/envs/common.py:
```diff
@@ -871,7 +871,6 @@
     'edx_django_utils.monitoring.MonitoringMemoryMiddleware',
 
     # Before anything that looks at cookies, especially the session middleware
```
You can remove the empty line and comment as well.
```suggestion
```

robrap commented on lms/envs/common.py:
```diff
@@ -2243,7 +2243,6 @@ def _make_locale_paths(settings):  # pylint: disable=missing-function-docstring
     'edx_django_utils.monitoring.FrontendMonitoringMiddleware',
 
     # Before anything that looks at cookies, especially the session middleware
```
You can remove the empty line and comment as well.
```suggestion
```

robrap reviewed (commented):
Thank you. Reminder that we won't merge until the DEPR is accepted.

1. When creating a PR, it is good practice to put a reference to the ticket it is implementing. For example: https://github.com/openedx/edx-platform/issues/36834
2. Please use "fix!: ..." for the commit title to designate that this is a breaking change, even though it is a breaking change that we don't think will affect anyone.
3. Please add a small commit description as well. Could be something like:
```
Removes the CookieNameChange middleware that was intended
to be temporary with a target removal date of October 1, 2021.
This middleware was designed for a transitional cookie migration
pattern that is no longer needed.

Implements DEPR:
- https://github.com/openedx/edx-platform/issues/36834
```

ktyagiapphelix2u committed:
fix: Remove deprecated CookieNameChange middleware

The CookieNameChange middleware has been removed. Originally a temporary measure slated for removal by October 1, 2021, it facilitated a transitional cookie migration pattern that is no longer required. This action addresses DEPR: https://github.com/openedx/edx-platform/issues/36834

ktyagiapphelix2u commented:
Removed the deprecated CookieNameChange middleware, which was originally introduced as a temporary solution for cookie migration, with a planned removal date of October 1, 2021. As the transitional pattern is no longer required, the middleware has been safely removed.

Also removed the associated inline comments as requested.

Implements DEPR:
- https://github.com/openedx/edx-platform/issues/36834

robrap commented:
@ktyagiapphelix2u: [inform] The additional PR comment was not needed. It is just redundant. Thank you for updating the commit comment here: https://github.com/openedx/edx-platform/pull/36815/commits/bb4000884668ed9eedaab3814ce34c6435e02e85.

robrap committed:
Merge branch 'master' into ktyagiapphelix2u/remove_cookie_name

robrap reviewed (approved):
The DEPR has been accepted. I'm waiting for tests to pass, and then I can merge this. Thank you @ktyagiapphelix2u.
